internal/ir/xds.go

```diff
@@ -1123,13 +1126,12 @@ func (h TCPListener) Validate() error {
 }
 
 // TLSInspectorConfig holds the configuration required for inspecting TLS
-// passthrough connections.
+// connections.
 // +k8s:deepcopy-gen=true
 type TLSInspectorConfig struct {
 	// Server names that are compared against the server names of a new connection.
 	// Wildcard hosts are supported in the prefix form. Partial wildcards are not
 	// supported, and values like *w.example.com are invalid.
-	// SNIs are used only in case of TLS Passthrough.
 	SNIs []string `json:"snis,omitempty" yaml:"snis,omitempty"`
```

Comment:
I think it's already exposed as `TLS.Passthrough` isn't it? I can add `SNIs` directly to `TLSConfig` if you prefer

Reply:
I'd vote for making `TLSInspectorConfig` an embedded field within `TLS` https://github.com/envoyproxy/gateway/blob/4d6da72fe523e732531fdfc02d499b0da3aaa185/internal/ir/xds.go#L1088 , removing the need for a dedicated `Passthrough` field

* passthrough logic can be computed using - `isTLSPassthrough := irListener.TLS != nil && irListener.TLS.Terminate == nil`
* this also eliminates the need to add `Inspector` into the `TLSConfig` field, which is common to HTTPS, which uses another field (`listener.Hostname`) for SNI inspection

Reply:
I pushed an update here, it changes a different part of the IR though, `passthrough` -> `inspector`, since that's now common between `mode: Terminate` and `mode: Passthrough`